Mage.Sets/src/mage/cards/e/EvolvingAdaptive.java

```diff
@@ -0,0 +1,99 @@
+package mage.cards.e;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.*;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.CountersSourceCount;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.BoostSourceEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.counters.CounterType;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+public class EvolvingAdaptive extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterCreaturePermanent("another creature");
+
+    static {
+        filter.add(AnotherPredicate.instance);
+    }
+
+    private static final DynamicValue oilCounters = new CountersSourceCount(CounterType.OIL);
+
+    public EvolvingAdaptive(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{G}");
+        this.addSubType(SubType.PHYREXIAN);
+        this.addSubType(SubType.WARRIOR);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(0);
+
+        //Evolving Adaptive enters the battlefield with an oil counter on it.
+        this.addAbility(new EntersBattlefieldAbility(
+                new AddCountersSourceEffect(CounterType.OIL.createInstance()),
+                "with an oil counter on it"
+        ));
+
+        //Evolving Adaptive gets +1/+1 for each oil counter on it.
+        this.addAbility(new SimpleStaticAbility(new BoostSourceEffect(oilCounters, oilCounters, Duration.WhileOnBattlefield)));
```

Comment:
Please apply `.setText("{this} gets +1/+1 for each oil counter on it")` to `BoostSourceEffect` for desired rule text

Reply:
done